Sources/ApexyAlamofire/AlamofireClient.swift

```diff
@@ -95,21 +95,24 @@ open class AlamofireClient: Client {
     /// - Returns: The progress of fetching the response data from the server for the request.
     open func request<T>(
         _ endpoint: T,
-        completionHandler: @escaping (APIResult<T.Content>) -> Void
+        completionHandler: @escaping (Result<T.Content, T.Failure>) -> Void
     ) -> Progress where T: Endpoint {
 
-        let anyRequest = AnyRequest(create: endpoint.makeRequest)
-        let request = sessionManager.request(anyRequest)
-            .validate { request, response, data in
-                Result(catching: { try endpoint.validate(request, response: response, data: data) })
-            }.responseData(
+        let urlRequestResult = endpoint.makeRequest()
+        guard case let .success(urlRequest) = urlRequestResult else {
+            if case let .failure(error) = urlRequestResult {
+                completionHandler(.failure(error))
+            }
+            return Progress()
```

Comment:
`func request` now have two exit points executes simultaneously: `return` and `completionHandler`, it code smells

Reply:
completion handler is not exit point at all. It smells but the only mistake here is that completion handler is called not in its queue